### DISCUSSION TITLE
minio not picking up CAs of other clients

### DISCUSSION THREAD

**Discussion by shivakumargowda12** [minio/minio#16825] [Q&A] [answered]
Hi, I have configured minio with SSL by following the blog - https://gist.github.com/fritsstegmann/21a76c40ea7642802ee8a49ed635fc31

While trying to setup OIDC with Keycloak Server which is running in HTTPS mode, facing below issue
The CAs of Keycloak Server provided as part of the same cert path is not working.
The login page errors out saying - **X509: Certificate Signed by Unknown Authority**

**Comment by klauspost** [comment, marked as answer]
Use the official docs, not a random gist put up 2 years ago.

https://min.io/docs/minio/linux/operations/network-encryption.html#enabling-tls

**Reply by shivakumargowda12** [reply]
Thanks for the reply. But we have went through this already for configuring TLS. Dint work. 
So, after searching many blogs, we found the above link which worked for setting up TLS.

But still CAs part dint work though.

**Reply by klauspost** [reply]
It literally says in the link above how to set the directory from which certificates are read. So either do that or provide more information than "Dint work".

**Reply by shivakumargowda12** [reply]
For ex: 
If we provide certs-dir as /home/minio/certs, it was not working. Only HTTP was working. 
It was expecting the path till parent directory only, that is only /home/minio, it started picking up the certificates and HTTPS started working

**Reply by shivakumargowda12** [reply]
The **CA certificate** that we had, dint had **Basic Constraints** field set.. Hence minio server was throwing error.
I felt, the error description could have been different if there is any error in the certificates.
Anyway, thanks for the help.

**Reply by klauspost** [reply]
I believe certificate loads errors just got improved.
